### ISSUE TITLE
[X86] Uninitialized variable warnings in cetintrin.h

### ISSUE THREAD

**Issue by RKSimon** [llvm/llvm-project#55224]
https://godbolt.org/z/bqxb7fdvr

We see these with "-Wall -Wsystem-headers" builds:

```
/opt/compiler-explorer/clang-assertions-trunk-20220502/lib/clang/15.0.0/include/cetintrin.h:47:32: warning: variable 't' is uninitialized when used here [-Wuninitialized]
  return __builtin_ia32_rdsspd(t);
                               ^
/opt/compiler-explorer/clang-assertions-trunk-20220502/lib/clang/15.0.0/include/cetintrin.h:46:17: note: initialize the variable 't' to silence this warning
  unsigned int t;
                ^
/opt/compiler-explorer/clang-assertions-trunk-20220502/lib/clang/15.0.0/include/cetintrin.h:57:32: warning: variable 't' is uninitialized when used here [-Wuninitialized]
  return __builtin_ia32_rdsspq(t);
                               ^
/opt/compiler-explorer/clang-assertions-trunk-20220502/lib/clang/15.0.0/include/cetintrin.h:56:23: note: initialize the variable 't' to silence this warning
  unsigned long long t;
                      ^
```
Noticed while triaging #35297

CC @phoebewang

**Comment by phoebewang** [comment]
Thanks @RKSimon for the reporting. In fact I verified it during review. Do you know why the warning only emitted when specify `-Wsystem-headers`?

**Comment by RKSimon** [comment]
Thanks @phoebewang

After f2b16488129ef408e89c900f61dc53b1a6699c45 we now test system headers (including target includes) with Wextra in clang/test/Headers/x86-intrinsics-headers-clean.cpp to check for this in the future.
